src/custom-apps/components/WebhookHeaders/WebhookHeadersTableBody.tsx

```diff
@@ -0,0 +1,118 @@
+import TableRowLink from "@dashboard/components/TableRowLink";
+import { FormChange } from "@dashboard/hooks/useForm";
+import { removeAtIndex, updateAtIndex } from "@dashboard/utils/lists";
+import { TableBody, TableCell, TextField } from "@material-ui/core";
+import { DeleteIcon, IconButton } from "@saleor/macaw-ui";
+import clsx from "clsx";
+import React, { ChangeEvent } from "react";
+import { useIntl } from "react-intl";
+
+import { messages } from "./messages";
+import useStyles from "./styles";
+import { Header, stringifyHeaders } from "./utils";
+
+const nameSeparator = ":";
+const nameInputPrefix = "name";
+const valueInputPrefix = "value";
+
+export interface WebhookHeadersTableBodyProps {
+  onChange: FormChange;
+  headers: Header[];
+}
+
+const WebhookHeadersTableBody: React.FC<WebhookHeadersTableBodyProps> = ({
+  onChange,
+  headers,
+}) => {
+  const classes = useStyles();
+  const intl = useIntl();
+
+  const change = ({ target }: ChangeEvent<HTMLTextAreaElement>) => {
+    const { name, value } = target;
+    const [field, index] = name.split(nameSeparator);
+
+    const item = headers[index];
+
+    // lowercase header name
+    if (field === nameInputPrefix) {
+      item[field] = value.toLowerCase();
+    } else {
+      item[field] = value;
+    }
+
+    onChange({
+      target: {
+        name: "customHeaders",
+        value: stringifyHeaders(
+          updateAtIndex(item, headers, parseInt(index, 10)),
+        ),
+      },
+    });
+  };
+
+  return (
+    <TableBody>
+      {headers.map((field, fieldIndex) => (
+        <TableRowLink data-test-id="field">
+          <TableCell className={clsx(classes.colName, classes.tableCell)}>
+            <TextField
+              InputProps={{
+                classes: {
+                  input: classes.input,
+                },
+              }}
+              inputProps={{
+                "aria-label": `${nameInputPrefix}${nameSeparator}${fieldIndex}`,
+              }}
+              name={`${nameInputPrefix}${nameSeparator}${fieldIndex}`}
+              fullWidth
+              onChange={change}
+              value={field.name}
+              error={field.error}
+              helperText={
+                (field.error && intl.formatMessage(messages.headerNameError)) ||
+                " "
+              }
+            />
+          </TableCell>
+          <TableCell className={clsx(classes.colValue, classes.tableCell)}>
+            <TextField
+              InputProps={{
+                classes: {
+                  input: classes.input,
+                },
+              }}
+              inputProps={{
+                "aria-label": `${valueInputPrefix}${nameSeparator}${fieldIndex}`,
+              }}
+              name={`${valueInputPrefix}${nameSeparator}${fieldIndex}`}
+              fullWidth
+              onChange={change}
+              value={field.value}
+              helperText={" "}
```

Comment:
```suggestion
              helperText=""
```

Reply:
@poulch it that case it's been done for purpose for render empty helper text to prevent inputs shift on error

![Feb-22-2023 11-31-15](https://user-images.githubusercontent.com/779644/220594961-e8aeb0dc-29de-4efd-aee1-56b93686b01f.gif)